pkg/scheduler/scheduler_test.go

```diff
@@ -18,23 +18,1232 @@ package scheduler
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-binding-2",
+					Namespace: "default",
+				},
+				Spec: workv1alpha2.ResourceBindingSpec{
+					Placement: &policyv1alpha1.Placement{
+						ClusterAffinities: []policyv1alpha1.ClusterAffinityTerm{
+							{
+								AffinityName: "affinity1",
+								ClusterAffinity: policyv1alpha1.ClusterAffinity{
+									ClusterNames: []string{"cluster1"},
+								},
+							},
+							{
+								AffinityName: "affinity2",
+								ClusterAffinity: policyv1alpha1.ClusterAffinity{
+									ClusterNames: []string{"cluster2"},
+								},
+							},
+						},
+					},
+				},
+			},
+			scheduleResults: []core.ScheduleResult{
+				{},
+				{
+					SuggestedClusters: []workv1alpha2.TargetCluster{
+						{Name: "cluster2", Replicas: 1},
+					},
+				},
+			},
+			scheduleErrors: []error{errors.New("first affinity failed"), nil},
+			expectError:    false,
+		},
+		{
+			name: "all affinities fail",
+			binding: &workv1alpha2.ResourceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-binding-fail",
+					Namespace: "default",
+				},
+				Spec: workv1alpha2.ResourceBindingSpec{
+					Placement: &policyv1alpha1.Placement{
+						ClusterAffinities: []policyv1alpha1.ClusterAffinityTerm{
+							{
+								AffinityName: "affinity1",
+								ClusterAffinity: policyv1alpha1.ClusterAffinity{
+									ClusterNames: []string{"cluster1"},
+								},
+							},
+							{
+								AffinityName: "affinity2",
+								ClusterAffinity: policyv1alpha1.ClusterAffinity{
+									ClusterNames: []string{"cluster2"},
+								},
+							},
+						},
+					},
+				},
+			},
+			scheduleResults: []core.ScheduleResult{{}, {}},
+			scheduleErrors:  []error{errors.New("first affinity failed"), errors.New("second affinity failed")},
+			expectError:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := karmadafake.NewSimpleClientset(tt.binding)
+			fakeRecorder := record.NewFakeRecorder(10)
+			mockAlgorithm := &mockAlgorithm{
+				scheduleFunc: func(_ context.Context, spec *workv1alpha2.ResourceBindingSpec, status *workv1alpha2.ResourceBindingStatus, _ *core.ScheduleAlgorithmOption) (core.ScheduleResult, error) {
+					index := getAffinityIndex(spec.Placement.ClusterAffinities, status.SchedulerObservedAffinityName)
+					if index < len(tt.scheduleResults) {
+						return tt.scheduleResults[index], tt.scheduleErrors[index]
+					}
+					return core.ScheduleResult{}, errors.New("unexpected call to Schedule")
+				},
+			}
+			s := &Scheduler{
+				KarmadaClient: fakeClient,
+				eventRecorder: fakeRecorder,
+				Algorithm:     mockAlgorithm,
+			}
+
+			err := s.scheduleResourceBindingWithClusterAffinities(tt.binding)
+
+			if (err != nil) != tt.expectError {
+				t.Errorf("scheduleResourceBindingWithClusterAffinities() error = %v, expectError %v", err, tt.expectError)
+			}
+		})
```

Comment:
The result is not judged later. The test method can be the same as that of the TestScheduleResourceBindingWithClusterAffinity.